libs/blocks/global-navigation/utilities/getUserEntitlements.js

```diff
@@ -63,27 +130,37 @@ const getSubscriptions = async ({ queryParams, locale }) => {
  * @param {object} object required params
  * @param {array} object.params array of name value query parameters [{name: 'Q', value: 'PARAM'}]
  * @param {object} object.locale {country: 'CH', language: 'de'}
+ * @param {string} object.format format function, raw or default
  * @returns {object} JIL Entitlements
  */
-const getUserEntitlements = async ({ params, locale } = {}) => {
-  if (!window.adobeIMS?.isSignedInUser()) return Promise.reject(new Error('User not signed in'));
+const getUserEntitlements = async ({ params, locale, format } = {}) => {
+  if (!window.adobeIMS?.isSignedInUser()) return Promise.resolve(emptyEntitlements());
 
   const queryParams = getQueryParameters(params);
-  if (entitlements[queryParams]) return entitlements[queryParams];
+  if (entitlements[queryParams]) {
+    return format === 'raw'
+      ? entitlements[queryParams]
+      : entitlements[queryParams]
+        .then((res) => mapSubscriptionCodes(res));
+  }
 
   let resolve;
-  let reject;
   entitlements[queryParams] = entitlements[queryParams]
-    || new Promise((_resolve, _reject) => {
-      resolve = _resolve;
-      reject = _reject;
-    });
-  setTimeout(() => reject('Subscriptions API Call timeout'), API_WAIT_TIMEOUT);
+   || new Promise((_resolve) => { resolve = _resolve; });
+
+  setTimeout(() => resolve(emptyEntitlements()), API_WAIT_TIMEOUT);
   // TODO we might need to format data for analytics
-  const data = await getSubscriptions({ queryParams, locale });
+  const data = await getSubscriptions({ queryParams, locale })
+    .finally((res) => res || emptyEntitlements());
```

Comment:
as per, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

finally does not take any argument, `res` is not useful here.

Reply:
Indeed, changed that 👍